src/Component.php

```diff
@@ -191,9 +194,16 @@ protected function getVariantDefaultVariables(string $modifier, array $defined_v
    *   3. a list of variants, if any.
    */
   protected function extractParts(string $compound_name): array {
-    $pathname = preg_replace('@--[^.]+@', '', $compound_name);
+    $library = Drupal::service('twig.loader.componentlibrary');
     $variants = explode('--', basename(basename($compound_name, '.twig'), '.html'));
     $component = array_shift($variants);
+    // If no file-based variant template file exist use the base component otherwise.
+    if (!$library->exists($compound_name)) {
+      $pathname = preg_replace('@--[^.]+@', '', $compound_name);
+    }
+    else {
+      $pathname = $compound_name;
```

Comment:
As the returned pathname is used for loading the component's definition/configuration file, and as variant-specific definition/configuration files are not supported (as documented), I don't understand why we would ever want to return the compound name (including variants) as pathname?

Reply:
As the PR title and the related issue states, we want to support file based variant files, but no dedicated configuration files. As you can see [here](https://github.com/netzstrategen/drupal-twig_fractal/blob/master/src/Node/Render.php#L68) the node is calling `$component->getPathname()` as the template value.

Reply:
As you can see [here](https://github.com/netzstrategen/drupal-twig_fractal/blob/master/src/Component.php#L60) the Component is calling `$component->getPathname()` to load the config file.

Looks like property overloading.  We probably need a separate method `getTemplatePathname()` then.

Reply:
I also changed the config loading code in this PR:

```diff
- return $path['dirname'] . '/' . $path['filename'] . '.config.yml';
+ return $path['dirname'] . '/' . $this->getName() . '.config.yml';
```

Changing the argument seems to be a more valid solution to me. 👍